Increase search results for hymns from 10 to 50

sukhpalsingh opened the pull request:
Adding the settings to select the number of results showing to the user when searching by any of the search type.

jasdeepkhalsa commented:
I'm all for increasing the number of results to 50 and keeping all your logic to keep this configurable / maintainable within the code. However, not so keen on having this as an option we offer to the user because:

* Generally well designed websites do not tend to let users decide how many results will be shown to them. Google is a great example. It's always 10 results per page because it allows them to keep searches fast whilst encouraging them to push for more optimal search algorithms. Users tend not to even visit page 2 or 3 of Google search results for this reason because they find what they're looking for. We should aim for the same level of user experience.
* It's a pretty destructive setting to provide to users. Instead, we should provide an optimum number of results that works great across all platforms (so far I've come to the conclusion of 50 being a good number) to avoid users getting into performance issues in the app with this as a setting & which they would ultimately blame us for
* We do not currently have pagination, but this may be something we do in the future which should solve the remaining issue of the full results not showing up, whilst we can still control the limit per page to keep the performance high
* We do not currently have filters for results, but this may also be something we introduce, which should help the user make their searches more specific whilst not needing to increase the number of results

jasdeepkhalsa commented:
Also, well done for this code, as its shows you're getting much more familiar with the code base

jasdeepkhalsa commented:
Also with the change for iOS, you also mentioned:

* Some build step you had to comment out? Maybe add this to your PR
* Did you test on iPad too? If not would be good to try this out if you can get access to one

sukhpalsingh commented:
I will test on iPad soon. Lost the charge for it thats why could not test last time. Once i get the charger sort-out it will then start testing on iPad also.

I remember commenting a line in Objective C class but can't remember where i did so. I do not think it is anything in our app code. It is some sort of plugin to hide/show keyboard and has name started something like "hacking keyboard". I will try to find the place where it happened and will submit the changes in another merge request for that.

Thanks for the review. I have removed the preference option for search limit. I also removed from all the code as it was making code more complicated which we do not need if we do not have any such preference.

jasdeepkhalsa commented:
I think this plugin may be the cordova plugin `com.ionic.keyboard` found in `package.json`? We may need to upgrade this if it's causing a problem, but it should work with iOS.

Also, I think `URLResolver` in `services.js` would also need to be modified to get the correct file paths for iOS? For example did you try Prayers on iOS? Technically this should not work! I'd be very surprised :)

sukhpalsingh commented:
Prayers are working fine on iOS. Haven't look at the URLResolver yet as it did not give any issue to me while testing on iPhone. I will check this out.

Icons for the app did not match the iOS specifications so i resized them when i built the app and will copy those icons somewhere in the images folder of the app so that we can automate this process to make the build easier. As i done this manually last time.

jasdeepkhalsa commented:
If it's not broke, we won't fix it then haha :)

Yep, definitely add those icons in

jasdeepkhalsa commented on www/js/services.js:
```diff
@@ -8,17 +8,15 @@ angular.module('starter.services', [])
-    else
-    {
-      return Store.get(store);
+      data = defaults;
```
Surely this file should be restored to it's original otherwise the defaults will always over-ride any customizations the user makes?

sukhpalsingh replied:
I will not override default as data = defaults only happens when the user do not have any preference set. I have refactor the way we was loading default options. Please see the whole code together and u will see the logic i have written now.

jasdeepkhalsa commented on www/js/services.js:
```diff
@@ -102,14 +100,14 @@ return {
         var field = field || 'gurmukhi_search';
-        var sql = sql || "SELECT * FROM scriptures WHERE "+field+" LIKE '"+query+"%' LIMIT 10";
+        var sql = sql || "SELECT * FROM scriptures WHERE "+field+" LIKE '"+query+"%' LIMIT 50";
         var params = '?filter='+field + ' like '+query+'%&limit=10';
```
This should also be 50 (affects Web only)

jasdeepkhalsa replied:
Perhaps extract out into a `var LIMIT = '50'` so we can just change in one place next time!

jasdeepkhalsa commented on www/js/services.js:
```diff
@@ -102,14 +100,14 @@ return {
         var field = field || 'translation';
-        var sql = sql || "SELECT * FROM scriptures WHERE "+field+" LIKE '%"+query+"%' LIMIT 10";
+        var sql = sql || "SELECT * FROM scriptures WHERE "+field+" LIKE '%"+query+"%' LIMIT 50";
         var params = '?filter='+field + ' like %'+query+'%&limit=10';
```
This should also be 50 (affects Web only)

jasdeepkhalsa commented on www/js/services.js:
```diff
@@ -8,17 +8,21 @@ angular.module('starter.services', [])
+    return data;
+})
+
+.factory('AppSettings', function() {
```
Nice addition!
